File: backend/controllers/moviesController.js
import axios from "axios";
import { getWatchProviders } from "../services/tmdbService.js";

//Buscar películas
export const searchMovie = async (req, res) => {
  const { q } = req.query;
  try {
    const response = await axios.get(
      "https://api.themoviedb.org/3/search/movie",
      {
        params: {
          api_key: process.env.TMDB_API_KEY,
          query: q,
          language: "es-ES",
        },
      }
    );
    res.json(response.data);
  } catch (error) {
    console.error("Error al buscar:", error.message);
    res.status(500).json({ error: "Error al buscar película" });
  }
};

//Obtener una película por ID
export const getMovieById = async (req, res) => {
  const { id } = req.params;
  try {
    const response = await axios.get(
      `https://api.themoviedb.org/3/movie/${id}`,
      {
        params: {
          api_key: process.env.TMDB_API_KEY,
          language: "es-ES",
        },
      }
    );
    res.json(response.data);
  } catch (error) {
    console.error("Error al obtener película:", error.message);
    res.status(500).json({ error: "Error al obtener película" });
  }
};

// Obtener plataformas (Netflix, Disney+, etc.)
export const getMovieProviders = async (req, res) => {
  const { id } = req.params;
  try {
    const providers = await getWatchProviders(id);
    res.json(providers);
  } catch (error) {
    console.error("Error al obtener plataformas:", error.message);
    res.status(500).json({ error: "Error al obtener plataformas" });
  }
};

File: backend/controllers/movies_controller.py
from flask import jsonify, request
import requests, os
from services.tmdb_service import get_watch_providers

API_KEY = os.getenv("TMDB_API_KEY")
BASE_URL = "https://api.themoviedb.org/3"


def list_movies():
    page = request.args.get("page", 1)
    genre = request.args.get("genre")
    params = {"api_key": API_KEY, "language": "es-ES", "page": page}
    if genre:
        params["with_genres"] = genre
    res = requests.get(f"{BASE_URL}/discover/movie", params=params)
    return jsonify(res.json())


def get_movie_by_id(id):
    res = requests.get(f"{BASE_URL}/movie/{id}", params={"api_key": API_KEY, "language": "es-ES"})
    return jsonify(res.json())


def search_movie():
    q = request.args.get("q", "").strip()
    if not q:
        return jsonify({"results": []}), 200
    res = requests.get(f"{BASE_URL}/search/movie", params={"api_key": API_KEY, "language": "es-ES", "query": q})
    return jsonify(res.json())


def popular_movies():
    res = requests.get(f"{BASE_URL}/movie/popular", params={"api_key": API_KEY, "language": "es-ES"})
    return jsonify(res.json())


def movie_platforms(id):
    providers = get_watch_providers(id)
    return jsonify(providers)
